Spigot-Server-Patches/0594-Damage-with-any-cause.patch

```diff
@@ -0,0 +1,521 @@
++                return DamageSource.BURN;
++            case MELTING:
++                return CraftEventFactory.MELTING;
++            case LAVA:
++                return DamageSource.LAVA;
++            case DROWNING:
++                return DamageSource.DROWN;
++            case VOID:
++                return DamageSource.OUT_OF_WORLD;
++            case LIGHTNING:
++                return DamageSource.LIGHTNING;
++            case STARVATION:
++                return DamageSource.STARVE;
++            case POISON:
++                return CraftEventFactory.POISON;
++            case MAGIC:
++                return DamageSource.MAGIC;
++            case WITHER:
++                return DamageSource.WITHER;
++            case FALLING_BLOCK:
++                return DamageSource.FALLING_BLOCK;
++            case THORNS:
++                if (attackingEntity == null) {
++                    return DamageSource.GENERIC;
++                }
++                return DamageSource.thorns(attackingEntity);
++            case DRAGON_BREATH:
++                return DamageSource.DRAGON_BREATH;
++            case CUSTOM:
++                return DamageSource.GENERIC;
++            case FLY_INTO_WALL:
++                return DamageSource.FLY_INTO_WALL;
++            case HOT_FLOOR:
++                return DamageSource.HOT_FLOOR;
++            case CRAMMING:
++                return DamageSource.CRAMMING;
++            case DRYOUT:
++                return DamageSource.DRYOUT;
++        }
++        return DamageSource.GENERIC;
++    }
++
++}
+diff --git a/src/main/java/net/minecraft/server/DamageSource.java b/src/main/java/net/minecraft/server/DamageSource.java
+index 6fe5678cffc2487fe00c953d772f764bb37a4b11..20174f25d4935e377aacaf2bbe26b3d0f44583a7 100644
+--- a/src/main/java/net/minecraft/server/DamageSource.java
++++ b/src/main/java/net/minecraft/server/DamageSource.java
+@@ -1,5 +1,9 @@
+ package net.minecraft.server;
+ 
++import org.bukkit.craftbukkit.event.CraftEventFactory;
++import org.bukkit.entity.LightningStrike;
++import org.bukkit.event.entity.EntityDamageEvent;
++
+ import javax.annotation.Nullable;
+ 
+ public class DamageSource {
+@@ -150,6 +154,24 @@ public class DamageSource {
+         this.translationIndex = s;
+     }
+ 
++    // Paper start - OBFHELPER
++    public static DamageSource thorns(Entity attackingEntity) {
++        return DamageSource.a(attackingEntity);
++    }
++
++    public static DamageSource witherSkull(EntityWitherSkull attackingEntity, IProjectile projectile) {
++        return DamageSource.a(attackingEntity, projectile.getShooter());
++    }
++
++    public static DamageSource fireworkRocket(EntityFireworks attackingEntity, IProjectile projectile) {
++        return DamageSource.a(attackingEntity, projectile.getShooter());
++    }
++
++    public static DamageSource trident(Entity attackingEntity, IProjectile projectile) {
++        return DamageSource.a(attackingEntity, projectile.getShooter());
++    }
```

Comment:
What's the difference? They're all the same?

Reply:
a method is overloaded